common/config/azure-pipelines/templates/core-build.yaml

```diff
@@ -28,6 +31,9 @@ steps:
     displayName: rush build:ci
     workingDirectory: ${{ parameters.workingDir }}
     condition: and(succeeded(), not(and(eq(variables['Build.Reason'], 'PullRequest'), eq(variables['Agent.OS'], 'Darwin'))))
+    env:
```

Comment:
i _think_ we need to push to the remote build cache in a separate step.

this step i believe runs on every commit to a pr when its a windows/linux agent.
and we probably only want to write to the cache from one agent? (idk, im not sure)

Reply:
Right. I am not sure we can push outside of the build step (at least not through Rush directly) - but we can control when to write via an ENV variable. Are you thinking we should write on PR merge builds only?

Reply:
Yea im thinking only on PR Merge, if we're contantly writing to the cache on every commit in a pr, and for every open pr i feel it could get messy. hopefully in the pr it would still pull from the build cache the last known good, and just build up the changed pkgs using the diff

Reply:
>  hopefully in the pr it would still pull from the build cache the last known good, and just build up the changed pkgs using the diff

Yep it should.

Reply:
What are the conditions for the PR merge? Run on master (or a release) and not manually triggered?

Reply:
yep, we're doing it somewhere already, ill try to dig it up, but might have to wait til mon :)